asr1k_neutron_l3/plugins/l3/service_plugins/l3_extension_adapter.py

```diff
@@ -300,6 +300,14 @@ def get_sync_data(self, context, router_ids=None, active=None, host=None):
                     if gw_info is not None:
                         gw_info['external_fixed_ips'] = gw_port['fixed_ips']
 
+            # add address scope to internal subnets
+            all_ports = router['_interfaces'] + ([gw_port] if gw_port else [])
+            all_subnets = {sn['id'] for port in all_ports for sn in port.get('subnets', [])}
+            subnet_address_scope_map = self.db.get_subnet_address_scope_map(context, all_subnets)
+            for port in all_ports:
+                for subnet in port.get('subnets', []):
+                    subnet['address_scope_id'] = subnet_address_scope_map.get(subnet['id'])
```

Comment:
Could it become a problem if the subnet is not present in the map and we set the key to None?
Or would guarding it with a check if its None make sense to not set the address_scope_id key in the subnet dictionary at all? I don't know if the key is mandatory, so we actually need to set it to something and I know my grep-foo isn't that good, looking through the neutron code - just asking to ensure this was considered.